redbot/core/events.py

```diff
@@ -99,10 +100,96 @@ async def on_ready():
                     "Outdated version! {} is available "
                     "but you're using {}".format(data["info"]["version"], red_version)
                 )
-                outdated_red_message = (
+                outdated_red_message = _(
                     "Your Red instance is out of date! {} is the current "
                     "version, however you are using {}!"
                 ).format(data["info"]["version"], red_version)
+                requires_python = data["info"]["requires_python"]
+                current_python = platform.python_version()
+                extra_update = _(
+                    "\n\nWhile the following should work in most scenarios as it is "
+                    "based on your current OS, environment, and Python version "
+                    "we highly recommend you to read the update docs at <{docs}> and "
+                    "make sure there is nothing else that "
+                    "needs to be done during the update. "
+                    "**Should the above not work for you,"
+                    " ensure you have read the update docs.**"
+                ).format(docs="https://docs.discord.red/en/stable/update_red.html",)
+                if expected_version(current_python, requires_python):
+                    package_extras = ""
+                    try:
+                        import alabaster
+                        import jinja2
+                        import requests
+                        import sphinx
+                        import urllib3
+                    except ImportError:
+                        docs = ""
+                    else:
+                        docs = "docs"
+                    try:
+                        import asyncpg
+                    except ImportError:
+                        postgres = ""
+                    else:
+                        postgres = "postgres"
+
+                    try:
+                        import black
+                        import regex
+                        import toml
+                        import typed_ast
+                    except ImportError:
+                        style = ""
+                    else:
+                        style = "style"
+
+                    try:
+                        import astroid
+                        import isort
+                        import pylint
+                        import pyparsing
+                        import pluggy
+                        import more_itertools
+                    except ImportError:
+                        test = ""
+                    else:
+                        test = "test"
+
+                    if any(i for i in [test, postgres, style, docs]):
+                        package_extras = "[{}]"
+                        valid = [i for i in [test, postgres, style, docs] if i]
+                        package_extras.format(",".join(valid))
```

Comment:
I have a better way for you ;)
```suggestion
                    installed_extras = []
                    for extra, reqs in red_pkg._dep_map:
                        if extra is None:
                            continue
                        try:
                            pkg_resources.require(req.name for req in reqs)
                        except pkg_resources.DistributionNotFound:
                            pass
                        else:
                            installed_extras.append(extra)

                    if installed_extras:
                        package_extras = f"[{','.join(installed_extras)}]"
                    else:
                        package_extras = ""
```
Yes, this does use a private attribute of `pkg_resources.Distribution` but it's way cleaner and less prone to omission in future when we add more extras or change some of the dependencies in the extras.
Also, this is used in this function a little below already so there's not much against using here too if it can help.